recipes/mender-client/debian-master/patches/boost-1.6.x-beast.patch

```diff
@@ -0,0 +1,132 @@
+Description: Patch the common/http implementation to go with Boost Beast 1.6.x
+Author: ole.orhagen@northern.tech
+Origin: other, <URL, required except if Author is present>
+Last-Update: 2023-05-11
+---
+This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
+Index: mender/common/http.hpp
+===================================================================
+--- mender.orig/common/http.hpp
++++ mender/common/http.hpp
+@@ -25,7 +25,7 @@
+ #ifdef MENDER_USE_BOOST_BEAST
+ #include <boost/asio.hpp>
+ #include <boost/beast.hpp>
+-#include <boost/beast/ssl.hpp>
++#include <boost/asio/ip/tcp.hpp>
+ #include <boost/asio/ssl/error.hpp>
+ #include <boost/asio/ssl/stream.hpp>
+ #endif // MENDER_USE_BOOST_BEAST
+@@ -48,6 +48,7 @@ namespace asio = boost::asio;
+ namespace beast = boost::beast;
+ namespace http = beast::http;
+ namespace ssl = asio::ssl;
++using tcp = boost::asio::ip::tcp; // from <boost/asio/ip/tcp.hpp>
+ #endif // MENDER_USE_BOOST_BEAST
+ 
+ namespace error = mender::common::error;
+@@ -314,7 +315,7 @@ private:
+ 
+ #ifdef MENDER_USE_BOOST_BEAST
+ 	boost::asio::ip::tcp::resolver resolver_;
+-	beast::ssl_stream<beast::tcp_stream> stream_;
++	ssl::stream<tcp::socket> stream_;
+ 
+ 	// This shared pointer is used as a workaround, points to ourselves, and has some peculiar
+ 	// properties. First the reason for the workaround: When calling `cancel()` on TCP streams,
+Index: mender/common/http/platform/beast/http.cpp
+===================================================================
+--- mender.orig/common/http/platform/beast/http.cpp
++++ mender/common/http/platform/beast/http.cpp
+@@ -14,7 +14,7 @@
+ 
+ #include <boost/asio/ip/tcp.hpp>
+ #include <boost/asio/ssl/verify_mode.hpp>
+-#include <boost/beast/core/stream_traits.hpp>
++#include <boost/beast/core/type_traits.hpp>
+ #include <boost/asio.hpp>
+ #include <common/http.hpp>
+ 
+@@ -94,7 +94,6 @@ Client::Client(ClientConfig &client, eve
+ 	resolver_(GetAsioIoContext(event_loop)),
+ 	stream_(GetAsioIoContext(event_loop), client.ctx_),
+ 	body_buffer_(HTTP_BEAST_BUFFER_SIZE) {
+-	response_buffer_.reserve(body_buffer_.size());
+ 
+ 	// Don't enforce limits. Since we stream everything, limits don't generally apply, and
+ 	// if they do, they should be handled higher up in the application logic.
+@@ -188,8 +187,8 @@ void Client::ResolveHandler(
+ 
+ 	weak_ptr<Client> weak_client(stream_active_);
+ 
+-	beast::get_lowest_layer(this->stream_)
+-		.async_connect(
++	boost::asio::async_connect(
++			this->stream_.next_layer(),
```

Comment:
Both of these approaches are available in later versions too. We could switch the main code to this. It's not really that bad.

Reply:
Keep in mind that they are both available, but are not really the same. Only in the case we have one layer wrapped.

Reply:
Yes, but we expect the next layer to be tcp. In fact, there is a different function one can use to get the lowest layer in 1.67, but this didn't work, which could mean that there is another, perhaps internal, layer after tcp. So I'd argue that `next_layer` is what we should use.

Reply:
Not against it. Just good to keep in mind for the future